### ISSUE TITLE
env create gives FutureWarning: `remote_definition` is deprecated

### ISSUE THREAD

**Issue by raybellwaves** [conda/conda#14355]
### Checklist

- [x] I added a descriptive title
- [x] I searched open reports and couldn't find a duplicate

### What happened?

I ran `conda env create -f environment-dev.yml -n myenv` and got

```
/opt/conda/lib/python3.12/argparse.py:2006: FutureWarning: `remote_definition` is deprecated and will be removed in 25.9. Use `conda env create --file=URL` instead.
  action(self, namespace, argument_values, option_string)
```

It also seems this message has popped up in the docs https://docs.conda.io/projects/conda/en/latest/commands/env/create.html under positional argumation

### Conda Info

```
(base) root@850a544f31cb:/# conda info

     active environment : base
    active env location : /opt/conda
            shell level : 1
       user config file : /root/.condarc
 populated config files : /opt/conda/.condarc
                          /root/.condarc
          conda version : 24.9.0
    conda-build version : 24.5.1
         python version : 3.12.7.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=x86_64
                          __conda=24.9.0=0
                          __glibc=2.31=0
                          __linux=6.6.51=0
                          __unix=0=0
       base environment : /opt/conda  (writable)
      conda av data dir : /opt/conda/etc/conda
  conda av metadata url : None
           channel URLs : HIDDEN
                          https://conda.anaconda.org/nodefaults/linux-64
                          https://conda.anaconda.org/nodefaults/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /opt/conda/pkgs
                          /root/.conda/pkgs
       envs directories : /opt/conda/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/24.9.0 requests/2.32.3 CPython/3.12.7 Linux/6.6.51-0-virt ubuntu/20.04.6 glibc/2.31 solver/libmamba conda-libmamba-solver/24.9.0 libmambapy/1.5.9
                UID:GID : 0:0
             netrc file : /root/.netrc
           offline mode : False
```
### Conda Config
```
channels:
  - conda-forge

==> /root/.condarc <==
channels:
  - https://HIDDEN.jfrog.io/artifactory/api/conda/conda/dtn
  - https://HIDDEN.jfrog.io/artifactory/api/conda/conda
  - nodefaults
show_channel_urls: True

==> envvars <==
allow_softlinks: False
```

**Comment by prady0t** [comment]
Hi! What should be the expected behaviour?

**Comment by cmspeed** [comment]
Think this is related to #14158 via #14160 implemented in version 24.9.0 (2024-09-26).
